Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm

```diff
@@ -525,6 +539,11 @@
     }
 
     postLayoutData.baseWritingDirection = frame.editor().baseWritingDirectionForSelectionStart();
+
+    if (!selection.isNone()) {
+        if (RefPtr editableRoot = selection.rootEditableElement())
```

Comment:
It seems the logic changed here a bit — previously, we would use `enclosingTextFormControl` (or otherwise fall back to `rootEditableElement`), but we're now only using the latter. Is that intentional?